src/hooks/useFormField.js

```diff
@@ -1,27 +1,52 @@
 import { useState } from "react";
-/*
-A custom hook controlling form field logic.
-It could be extended with:
-    - validation support (a function with validation rules could be passed as an argument)
-    - error message states 
-*/
+import { validateMultipleRules } from "../utils/ValidateRule/ValidateRule";
 
-export default function useFormField(initialState) {
-  const [currentState, setState] = useState(initialState);
+/**
+ * Hook to handle fields and validate them.
+ *
+ * @param {string} initialState - The intial value for the input field.
+ * @param {InputValidator} InputValidator - ValidatorRule that is used to validate the field.
+ *
+ */
 
-  function handleInputChange(e) {
-    switch (e.target.type) {
-      case "checkbox":
-        setState(e.target.checked);
-        break;
-      default:
-        setState(e.target.value);
-    }
-  }
+export default function useFormField(initialState, InputValidator) {
+  const NO_INPUT_VALIDATION = "Input validation is undefinied.";
+
+  const [currentState, setState] = useState(initialState);
+  const [inputValidation, setInputValidation] = useState(InputValidator);
```

Comment:
it would be nice if there is an alignment between the param name (`validator`) and the local var name (`validation`). I personally think I prefer the `validator` terminology. And probably we could use it everywhere and talk everywhere about validators: `EmailValidator`, `NonEmptyValidator`. `MinimumLengthValidator`, etc.

Reply:
Alright, let's go with Validator across the board!

Reply:
Agree with Mircea